client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/query/RemoteQuery.java

```diff
@@ -34,13 +36,23 @@
       super(queryFactory, queryString);
       this.cache = cache;
       this.serializationContext = serializationContext;
+      this.indexedQueryMode = IndexedQueryMode.FETCH;
+   }
+
+   RemoteQuery(QueryFactory queryFactory, RemoteCacheImpl<?, ?> cache, SerializationContext serializationContext,
+               String queryString, IndexedQueryMode indexQueryMode) {
+      super(queryFactory, queryString);
+      this.cache = cache;
+      this.serializationContext = serializationContext;
+      this.indexedQueryMode = indexQueryMode;
    }
 
    RemoteQuery(QueryFactory queryFactory, RemoteCacheImpl cache, SerializationContext serializationContext,
```

Comment:
It's a good opportunity to fix the cache param here; should be ```RemoteCacheImpl<?, ?>```

Reply:
ok